tests/gha-run-k8s-common.sh

```diff
@@ -321,7 +321,7 @@ function _get_k0s_kubernetes_version_for_crio() {
 	#
 	# The CRI-O repo for such version of Kubernetes expects something like:
 	# 1.27
-	k0s_version=$(curl -sSLf "https://docs.k0sproject.io/stable.txt")
+	k0s_version="1.31.5"
```

Comment:
```suggestion
	k0s_version="1.31.5+k0s.0"
```

to please the pattern substitution below and return 1.31 instead of 1.31.5.

Reply:
A possible improvement could be to ensure `deploy_k0s` and `_get_k0s_kubernetes_version_for_crio` use the same k0s version.

Reply:
Yep - now the PoC suggested that this pining approach is viable I've reworked this PR to do it (more) properly!